### ISSUE TITLE
Import WordPress database

### ISSUE THREAD

**Issue by aks4896** [Interlock-Rochester/interlock-rochester.github.io.src#1]
The native importer (http://docs.getpelican.com/en/stable/importer.html) for pelican works with:
- WordPress XML export
- RSS/Atom feed

I'm thinking we can ether:
- Create local WP instance and import from that
- Find another tool

**Comment by deejoe** [comment]
As per interlock-rochester/interlock-rochester.github.io#1 I had Pelican running from the server last night. I backed out of that config to serve the wp-engine version again briefly. I got that running by restoring an old copy of index.php from another directory into this our current bogus document root. 

I set up a SOCKS proxy via ssh -D 223344 our.web.server -p $PORT and then pointed my local browser to 223344 so I had a secure connection to the server as localhost. Then, I logged in to wordpress and ran the export. I did this out of concern user credentials might be included in the export.

I installed a new, separate instance of pelican with its unique dependencies (BS4, lxml, feedparser) and  ran pelican-import over the xml file downloaded (via the ssh proxy) from the wordpress instance. I now have a output/ directory with hundreds of .rst files, some of them our legit old posts and some of them clearly spam content. I'll try to find a way to share them non-publically to get help sorting the ham from the spam. 

The Export file doesn't export media, so that will also have to be handled. Once I have a way of sharing around the directory contents for help with sorting, I'll work on a way to script the modification of the legitimate .rst files so that their image anchors point to an image directory in a Pelican instance and not back to the wordpress instance.  

I'll open a separate issue for each of those items:  ham/spam sifting and media link porting.

Since this isssue is for importing, I'm going to keep it open until we get there.

**Comment by aks4896** [comment]
Thanks for doing this Joe.

Should we just host these files on a virtual machine at Interlock and work from there?

**Comment by antitree** [comment]
I can dockerize them with my other wordpress instances if that's easier

**Comment by aks4896** [comment]
Joe sent a link to the Interlock mailing list.

**Comment by deejoe** [comment]
I've gone ahead and put stuff in ernestine:/home/dzho/wpimport for those of
you who have accounts on ernestine and still have control of the private key
that gets you into it ;-)

If, that is, that's easier for you.

I've also put a copy of the uploads directory, as wp-import_uploads but as
images I think I can scan through them for go/no-go pretty quickly.

**Comment by deejoe** [comment]
An update: Getting there. 

I'm working through the .rst files that pelican-import auto-generated. I've tweaked all of them to replace absolute URLs to assets hosted on the old wordpress instance into relative links for copies of those files hosted in the pelican root's content/wp-uploads directory. (We can go ahead and put new image content in content/images but I thought it best to keep the import stuff separate).

I couldn't push to this the interlock-rochester/interlock-rochester.github.io.src repo directly, so forked my own deejoe/interock-rochester.github.io.src repo and have pushed my wp-import working branch to it. I've invited Alex's account to collaborate on that. It might work better in the long run to add me as a collaborator to this fork, I don't know.

My wp-import branch has a couple of shell one-liners saved as .sh files, one does the search and replace on all .rst files to fix the URL and some escaping problem. The other runs pelican on the .rst files one at a time and saves the good ones. The branch also includes those files that have been hand-edited so far to work, and a list of the files that didn't work initially, with annotations for those that have been fixed so far.

The branch is also pretty beefy, since it introduces the assets from the wp uploads directory. I think they're all legit and not spam but keep an eye out. It's about a half-gig in all.

Anyway, still working on it, but have gotten far enough that I wanted to push and give an update.
